llvm/lib/IR/ConstantFold.cpp

```diff
@@ -1461,17 +1411,13 @@ static ICmpInst::Predicate evaluateICmpRelation(Constant *V1, Constant *V2,
       [[fallthrough]];
     case Instruction::UIToFP:
     case Instruction::SIToFP:
-    case Instruction::ZExt:
```

Comment:
This code puzzles me. Not so much your change to it, but...

The method is called `evaluateICmpRelation`, and a whole lot of comments talk about how we can't evaluate floating point values. So how can we ever get the `UIToFP` or `SIToFP` cases here?

Reply:
I agree that this doesn't make sense. I'll clean it up in a followup change.